<!-- Open both an Ensenso stereo camera and a mono camera within the same
nodelet manager and run the color_point_cloud Python node. -->

<launch>
  <arg name="serial_stereo" doc="Serial of the stereo camera (default: first found stereo camera)" default="" />
  <arg name="serial_mono" doc="Serial of the mono mamera (default: first found mono camera)" default="" />
  <arg name="settings_stereo" doc="JSON file with stereo camera parameters" default="" />
  <arg name="settings_mono" doc="JSON file with mono camera parameters" default="" />
  <arg name="stereo_ns" doc="Namespace of the stereo camera" default="stereo" />
  <arg name="mono_ns" doc="Namespace of the mono camera" default="mono" />
  <arg name="camera_frame" doc="The stereo camera's tf frame (default: optical_frame_SERIAL_STEREO)" default="" />
  <arg name="link_frame" doc="The stereo camera's link tf frame (default: camera_frame)" default="" />
  <arg name="target_frame" doc="The tf frame the data will be returned in (default: camera_frame)" default="" />
  <arg name="tcp_port" doc="The TCP port to open on the NxLib (-1=off, 0=autoSelect, >0=portNumber)" default="-1" />
  <arg name="threads" doc="The number of threads used by the NxLib for this node (-1=autoDetect)" default="-1" />
  <arg name="wait_for_camera" doc="Whether this node should wait for the camera to become available" default="false" />

  <include file="$(find ensenso_camera)/launch/mono_stereo_nodelets.launch">
    <arg name="serial_stereo" value="$(arg serial_stereo)" />
    <arg name="serial_mono" value="$(arg serial_mono)" />
    <arg name="settings_stereo" value="$(arg settings_stereo)" />
    <arg name="settings_mono" value="$(arg settings_mono)" />
    <arg name="stereo_ns" value="$(arg stereo_ns)" />
    <arg name="mono_ns" value="$(arg mono_ns)" />
    <arg name="camera_frame" value="$(arg camera_frame)" />
    <arg name="link_frame" value="$(arg link_frame)" />
    <arg name="target_frame" value="$(arg target_frame)" />
    <arg name="tcp_port" value="$(arg tcp_port)" />
    <arg name="threads" value="$(arg threads)" />
    <arg name="wait_for_camera" value="$(arg wait_for_camera)" />
  </include>

  <node pkg="ensenso_camera" type="color_point_cloud" name="color_point_cloud">
    <param name="mono_serial" type="string" value="$(arg serial_mono)" />
    <param name="stereo_ns" type="string" value="$(arg stereo_ns)" />
    <param name="mono_ns" type="string" value="$(arg mono_ns)" />
  </node>
</launch>